robot-server/robot_server/protocols/protocol_models.py

```diff
@@ -1,23 +1,72 @@
 """Protocol file models."""
 from __future__ import annotations
 from datetime import datetime
-from pydantic import Field
+from pydantic import BaseModel, Extra, Field
 from typing import Sequence
 
 from opentrons.protocol_runner import ProtocolFileType
 from robot_server.service.json_api import ResourceModel
 from .analysis_models import ProtocolAnalysis
 
 
+class Metadata(BaseModel):
+    """Extra, nonessential information about the protocol.
+
+    This can include things like:
+
+    * A human-readable title and description.
+    * A last-modified date.
+    * A list of authors.
+
+    The exact set of fields is not defined by this API.
+    A protocol can define whatever it likes as its metadata,
+    and this API will pass it along without modification.
+    When reading a protocol's metadata,
+    you should treat it as an arbitrary JSON object from an untrusted source.
+
+    This metadata does *not* include
+    anything material to how the robot will run the protocol.
+    For example, it does not include the protocol's pipette and labware requirements.
+
+    (Exception:
+    Python protocols define their `apiLevel`,
+    which *does* affect how they're run,
+    as part of their metadata.
+    This is for historical reasons;
+    `apiLevel` is not truly metadata and doesn't really belong here.
+    You should not rely on a Python protocol's correct `apiLevel` being present here.
+    If a compelling need arises, this API may expose `apiLevel` some other way.)
+    """
```

Comment:
I might consider paring this down. Detail is good, but brevity is also really important to make sure people can (and will) read it. Also, given discussions today about #8334, I don't think it's a good idea to commit to `The exact set of fields is not defined by this API.` right now

```suggestion
    """Extra, nonessential information about the protocol.

    This can include data like:

    * A human-readable title and description.
    * A last-modified date.
    * A list of authors.

    Metadata may contain fields other than those explicitly
    listed in this schema. Any such fields should be treated
    as arbitrary.

    The metadata *should not* include information needed
    to run the protocol correctly. For historical reasons, Python
    protocols define their `apiLevel` inside their metadata, but
    this should be considered an exception to the rule.
    """
```